app/src/main/java/com/sun/doitpat/widget/WidgetService.kt

```diff
@@ -0,0 +1,9 @@
+package com.sun.doitpat.widget
+
+import android.content.Intent
+import android.widget.RemoteViewsService
+
+class WidgetService : RemoteViewsService() {
+    override fun onGetViewFactory(intent: Intent?): RemoteViewsFactory =
+            WidgetRemoteViewFactory(this.applicationContext, intent!!)
```

Comment:
intent?.let {
    WidgetRemoteViewFactory(applicationContext, it)
}

Reply:
Dùng let sẽ cần trả về trong trường hợp null nữa a ạ. Lúc đấy đằng nào cũng phải trả về null intent. E để !! Luôn